packages/ERTP/CONTRIBUTING.md

```diff
@@ -4,44 +4,42 @@ Thank you!
 
 ## Contact
 
-We use github issues for all bug reports: https://github.com/Agoric/ERTP/issues
+We use github issues for all bug reports:
+https://github.com/Agoric/agoric-sdk/issues Please add an [ERTP]
+prefix to the title and ERTP tag to ERTP-related issues.
 
 ## Installing, Testing
 
 You'll need Node.js version 11 or higher. 
 
-* git clone https://github.com/Agoric/ERTP/
-* npm install
-* npm test
+* git clone https://github.com/Agoric/agoric-sdk/
+* cd packages/ERTP
+* yarn install
```

Comment:
To get the intra-monorepo dependencies to point at each other, the `yarn install` has to be done at the top level. Doing it within `packages/ERTP/` will get us stuff from NPM. Not sure if that's what you want here or not.

Reply:
Oh, this is very interesting. Is there documentation on this part in particular? The reason I ask is that when I clone a new copy of `agoric-sdk` and then do `yarn install` in one of the packages, it appears to do the linkage. There's nothing in node_modules for the package except for `.bin` and `.cache` and the symlinks are there (this is a partial list from running `find . -type l -ls` in `agoric-sdk`):

```
32704943        0 lrwxr-xr-x    1 katelynsills     staff                  18 Dec  4 16:25 ./node_modules/@agoric/zoe -> ../../packages/zoe
32704939        0 lrwxr-xr-x    1 katelynsills     staff                  39 Dec  4 16:25 ./node_modules/@agoric/default-evaluate-options -> ../../packages/default-evaluate-options
32704942        0 lrwxr-xr-x    1 katelynsills     staff                  22 Dec  4 16:25 ./node_modules/@agoric/marshal -> ../../packages/marshal
32704934        0 lrwxr-xr-x    1 katelynsills     staff                  34 Dec  4 16:25 ./node_modules/@agoric/acorn-eventual-send -> ../../packages/acorn-eventual-send
32704941        0 lrwxr-xr-x    1 katelynsills     staff                  23 Dec  4 16:25 ./node_modules/@agoric/swingset-vat -> ../../packages/SwingSet
32704937        0 lrwxr-xr-x    1 katelynsills     staff                  19 Dec  4 16:25 ./node_modules/@agoric/ertp -> ../../packages/ERTP
32704945        0 lrwxr-xr-x    1 katelynsills     staff                  20 Dec  4 16:25 ./node_modules/@agoric/captp -> ../../packages/captp
32704936        0 lrwxr-xr-x    1 katelynsills     staff                  23 Dec  4 16:25 ./node_modules/@agoric/evaluate -> ../../packages/evaluate
32704935        0 lrwxr-xr-x    1 katelynsills     staff                  30 Dec  4 16:25 ./node_modules/@agoric/cosmic-swingset -> ../../packages/cosmic-swingset
32704944        0 lrwxr-xr-x    1 katelynsills     staff                  38 Dec  4 16:25 ./node_modules/@agoric/transform-eventual-send -> ../../packages/transform-eventual-send
32704940        0 lrwxr-xr-x    1 katelynsills     staff                  28 Dec  4 16:25 ./node_modules/@agoric/bundle-source -> ../../packages/bundle-source
32704938        0 lrwxr-xr-x    1 katelynsills     staff                  28 Dec  4 16:25 ./node_modules/@agoric/eventual-send -> ../../packages/eventual-send
```

Am I just misunderstanding something? Apologies for all the questions - I'm trying to deeply understand how this works.

Reply:
When I try to run a test making a change in a local package, I get this error, so maybe something different is going on?

```
[/Users/katelynsills/code/temp/agoric-sdk/packages/SwingSet/src/controller.js:1
Error: Cannot find module '@agoric/default-evaluate-options'. Please verify that the package.json has a valid "main" entry] {
  code: 'MODULE_NOT_FOUND',
  path: '/Users/katelynsills/code/temp/agoric-sdk/node_modules/@agoric/default-evaluate-options/package.json',
  requestPath: '@agoric/default-evaluate-options'
}
```

Reply:
> When I try to run a test making a change in a local package, I get this error, so maybe something different is going on?

You really need to run `yarn install` and `yarn build` from the top of the monorepo, or else the bundled files are not available for the subpackages to find.

Reply:
Ah, good point.

Reply:
@warner, I did some testing, and it looks like doing `yarn install` in any of the package directories does the same thing as doing `yarn install` from root.  (This isn't for the documentation, but just to make sure I understand how it works.)

Here are the steps I took:
```sh
mkdir temp
cd temp
git clone https://github.com/Agoric/agoric-sdk/
cd agoric-sdk/packages/zoe
# from inside /zoe:
yarn install 
cd ../ERTP
touch experiment.js
# add `export const experiment = 'this is the new version';` to experiment.js
cd ../zoe
```

add test in /packages/zoe to import { experiment } from
'@agoric/ertp/experiment';

```
test.only('experiment', t => {
  try {
    t.equals(experiment, 'old version');
  } catch (e) {
    t.assert(false, e);
  } finally {
    t.end();
  }
});
```
then: 
```
# go to /packages OR root
cd ..
yarn build
# should fail with a mismatch of actual vs expected
yarn test

# change the test to expect 'this is the new version'. 
# `yarn test` passes
```

Reply:
Huh, ok, good to know. I just looked at the docs, and they're annoyingly silent about the matter, except for one offhanded comment in a [blog post](https://yarnpkg.com/en/docs/workspaces#toc-how-to-use-it), that says "Finally, run yarn install somewhere, ideally inside the workspace root.". But I guess it works either way. It must be walking up the directory tree and looking for a `package.json` that cites the current directory as a workspace.

But yeah, `build` is not quite as built-in as `install`, so it *must* be run from the top level, or at least it must be run in all dependencies first.